compiler/src/iree/compiler/Codegen/Common/TileAndDistributeToWorkgroupsPass.cpp

```diff
@@ -113,105 +107,117 @@ static LogicalResult lowerDispatchWorkgroupCountFromDagRootOp(
   tileSizes.assign(rootOpConfig.getTileSizeVals(0));
   interchange.assign(rootOpConfig.getTileInterchangeVals(0));
 
-  // Resize tile sizes to the number of loops setting inner loops to 0.
-  tileSizes.resize(workloadValues.size(), 0);
-  // Check that the interchange vector is also equal to the number of loops
-  if (!interchange.empty()) {
-    if (interchange.size() < workloadValues.size()) {
-      auto seq = llvm::seq<int64_t>(interchange.size(), workloadValues.size());
-      interchange.append(seq.begin(), seq.end());
-    }
-    interchange.resize(workloadValues.size());
-  }
-  // For now assert that number of partitionable loops are less than the
-  // supported max.
-  // TODO(ravishankarm): Relax this restriction.
-  if (partitionableLoops.size() > kNumMaxParallelDims) {
-    return workgroupCountOp.emitOpError(
-               "expected number of partitionable loops to be less than or "
-               "equal to ")
-           << kNumMaxParallelDims;
-  }
-
-  MLIRContext *context = workgroupCountOp.getContext();
-  OpBuilder builder(context);
-  // Add as many arguments as the number of loops
-  Location loc = workgroupCountOp.getLoc();
-
-  // AffineMap for the number of workgroups = ceilDiv(workload, tileSize)
-  SmallVector<Value> numTiles;
-  numTiles.reserve(workloadValues.size());
-  builder.setInsertionPoint(workgroupCountOp);
-  Value one = builder.create<arith::ConstantIndexOp>(loc, 1);
-  llvm::DenseSet<unsigned> partitionableLoopsSet;
+  // Set tile sizes of non-partitioned loops to 0.
+  llvm::SmallDenseSet<unsigned> partitionableLoopsSet;
   partitionableLoopsSet.insert(partitionableLoops.begin(),
                                partitionableLoops.end());
+  for (auto loopId : llvm::seq<unsigned>(0, tileSizes.size())) {
+    if (partitionableLoopsSet.count(loopId)) continue;
```

Comment:
nit: I think `if (!partitionableLoopsSet.count(loopId)) tileSizes[loopId] = 0;` is more readable.

Reply:
Its more natural for me the other way round. Leaving as is since it is classified as a`nit`.